### PR TITLE
Account History for Liquidity Pools

### PR DESCRIPTION
**Pull request by josefleventon**
- [x] Fetch pools
- [x] Add liquidity [`AmmJoinPoolRequest`](https://github.com/elys-network/bindings/blob/13183c2d425b53f60149a7c50dc60ce7ace11202/bindings/src/trade_shield/msg/execute_msg.rs#L81-L86)
- [x] Withdraw liquidity [`AmmExitPoolRequest`](https://github.com/elys-network/bindings/blob/13183c2d425b53f60149a7c50dc60ce7ace11202/bindings/src/trade_shield/msg/execute_msg.rs#L87-L92)
- [x] Claim Rewards [`ClaimRewardsRequest`](https://github.com/elys-network/bindings/blob/13183c2d425b53f60149a7c50dc60ce7ace11202/bindings/src/trade_shield/msg/execute_msg.rs#L75-L78)

### CODE REVIEW

File: bindings/src/account_history/types/account_snapshot.rs
Comment (politeWall) [inline bindings/src/account_history/types/account_snapshot.rs:1]: if you change the the state structure you have to increment the HISTORY namespace so we don't get conflict type at migrate

File: .vscode/settings.json
Comment (politeWall) [inline .vscode/settings.json:1]: could you remove the `.vscode` directory

**Review by cosmic-vagabond — APPROVED** [review]
sounds good @josefleventon could you also share some test result using the script command you added thanks

**Comment by josefleventon** [comment]
## Fetch Pools
```zsh
$ elysd q --output json --node "tcp://localhost:26657" wasm contract-state smart "elys17p9rzwnnfxcjp32un9ug7yhhzgtkhvl9jfksztgw5uh69wac2pgs98tvuy" '{
        "get_liquidity_pools": {
            "filter_type": "filter_all"
        }
    }' | jq
{
  "data": {
    "pools": [
      {
        "assets": [
          {
            "token": {
              "denom": "ibc/2180E84E20F5679FCC760D8C165B60F42065DEF7F46A72B447CFF1B7DC6C0A65",
              "amount": "715000000"
            },
            "weight": "10737418240"
          },
          {
            "token": {
              "denom": "ibc/27394FB092D2ECCD56123C74F36E4C1F926001CEADA9CA97EA622B25F41E5EB2",
              "amount": "80000000"
            },
            "weight": "10737418240"
          }
        ],
        "pool_ratio": "50 : 50",
        "rewards_apr": "0",
        "borrow_apr": "0",
        "leverage_lp": "0",
        "perpetual": "0",
        "tvl": "0",
        "rewards": "0"
      },
      {
        "assets": [
          {
            "token": {
              "denom": "ibc/2180E84E20F5679FCC760D8C165B60F42065DEF7F46A72B447CFF1B7DC6C0A65",
              "amount": "86796655287"
            },
            "weight": "10737418240"
          },
          {
            "token": {
              "denom": "ibc/E2D2F6ADCC68AA3384B2F5DFACCA437923D137C14E86FB8A10207CF3BED0C8D4",
              "amount": "6363716196"
            },
            "weight": "10737418240"
          }
        ],
        "pool_ratio": "50 : 50",
        "rewards_apr": "0",
        "borrow_apr": "0",
        "leverage_lp": "0",
        "perpetual": "0.755590298188652271",
        "tvl": "152694.346297134937840368",
        "rewards": "0"
      },
      {
        "assets": [
          {
            "token": {
              "denom": "ibc/2180E84E20F5679FCC760D8C165B60F42065DEF7F46A72B447CFF1B7DC6C0A65",
              "amount": "137658497074"
            },
            "weight": "107374182400"
          },
          {
            "token": {
              "denom": "uelys",
              "amount": "32316057737"
            },
            "weight": "107374182400"
          }
        ],
        "pool_ratio": "50 : 50",
        "rewards_apr": "0",
        "borrow_apr": "0",
        "leverage_lp": "0",
        "perpetual": "1",
        "tvl": "275316.994148",
        "rewards": "0"
      }
    ]
  }
}
```

**Pull request merged**